app/services/determine_sign_off_period.rb

```diff
@@ -40,5 +40,10 @@ def census_range
     start_date..end_date
   end
 
-  delegate :in_performance_profile_range?, to: :previous_academic_cycle
+  def in_performance_profile_range?(current_date)
+    result = previous_academic_cycle.in_performance_profile_range?(current_date)
+    result = false if result && provider.present? && provider.performance_profile_signed_off?
```

Comment:
Feels like this extra logic is a bit hidden in this method (and class) with no indication of it. But I can't see an obvious way to improve it without a lot of changes. 🤔

Reply:
I have moved it out for now, it seems the lesser evil, until we re-encounter this weird monster later.

https://github.com/DFE-Digital/register-trainee-teachers/pull/4849/commits/e6e3c558322d64aed6ee0c7e7877883cbd356470